src/main/java/aquarium/shop/basket/Basket.java

```diff
@@ -0,0 +1,95 @@
+package aquarium.shop.basket;
+
+import aquarium.shop.product.Product;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+
+public class Basket {
+    private static Basket INSTANCE;
+
+    private Basket() {
+    }
+
+    private HashMap<Product, Integer> content = new HashMap<>();
+    private double wholePrice;
+
+    public HashMap<Product, Integer> getContent() {
+        return content;
+    }
+
+    public static Basket getInstance() {
+        if (INSTANCE == null) {
+            INSTANCE = new Basket();
+        }
+        return INSTANCE;
+    }
+
+    public double getWholePrice() {
+        return wholePrice;
+    }
+
+    public void addProduct(Product product) {
+        if (content.containsKey(product)) {
+            content.put(product, content.get(product) + 1);
+        } else {
+            content.put(product, 1);
+        }
+
+    }
+
+    public void addProduct(Product product, int quantity) {
+        if (content.containsKey(product)) {
+            content.put(product, content.get(product) + quantity);
+        } else {
+            content.put(product, quantity);
+        }
+    }
+
+    public void removeProduct(Product product) {
+        content.remove(product);
+    }
+
+    public void increaseNumberOfProducts(Product product) {
+        content.replace(product, content.get(product) + 1);
+    }
+
+    public void decreaseNumberOfProducts(Product product) {
+        content.replace(product, content.get(product) - 1);
+    }
+
+    public void clearBasket() {
+        content.clear();
+    }
+
+    public double calculateBasketPrice() {
```

Comment:
I believe that it would be better to update whole price after adding / removing project. You can just change actual **wholePrice** value accodring to added / removed products and in that approach you don't need to do anything with wholePrice when **showBasket** is executed, just return it.
Thanks that showBasket will be faster and recalculating logic won't be needed at all.